pkg/deviceshifu/deviceshifuopcua/deviceshifuopcua.go

```diff
@@ -164,57 +165,128 @@ type DeviceCommandHandlerOPCUA struct {
+	ctx := context.Background()
+	ctx, cancel := context.WithTimeout(ctx, time.Duration(*handler.timeout)*time.Millisecond)
+	defer cancel()
 
-		if err != nil {
-			http.Error(w, "Failed to read message from Server, error: "+err.Error(), http.StatusBadRequest)
-			logger.Errorf("Read failed: %s", err)
-			return
-		}
+	resp, err := handler.client.ReadWithContext(ctx, req)
+	handlerInstruction := handler.HandlerMetaData.instruction
 
-		if resp.Results[0].Status != ua.StatusOK {
-			http.Error(w, "OPC UA response status is not OK "+fmt.Sprint(resp.Results[0].Status), http.StatusBadRequest)
-			logger.Errorf("Status not OK: %v", resp.Results[0].Status)
-			return
-		}
+	if err != nil {
+		http.Error(w, "Failed to read message from Server, error: "+err.Error(), http.StatusBadRequest)
+		logger.Errorf("Read failed: %s", err)
+		return
+	}
 
-		logger.Infof("%#v", resp.Results[0].Value.Value())
-
-		w.WriteHeader(http.StatusOK)
-		// TODO: Should handle different type of return values and return JSON/other data
-		// types instead of plain text
-		rawRespBody := resp.Results[0].Value.Value()
-		rawRespBodyString := fmt.Sprintf("%v", rawRespBody)
-		respString := rawRespBodyString
-		instructionFuncName, shouldUsePythonCustomProcessing := deviceshifubase.CustomInstructionsPython[handlerInstruction]
-		logger.Infof("Instruction %v is custom: %v", handlerInstruction, shouldUsePythonCustomProcessing)
-		if shouldUsePythonCustomProcessing {
-			logger.Infof("Instruction %v has a python customized handler configured.\n", handlerInstruction)
-			respString = utils.ProcessInstruction(deviceshifubase.PythonHandlersModuleName, instructionFuncName, rawRespBodyString, deviceshifubase.PythonScriptDir)
-		}
-		fmt.Fprintf(w, "%v", respString)
+	if resp.Results[0].Status != ua.StatusOK {
+		http.Error(w, "OPC UA response status is not OK "+fmt.Sprint(resp.Results[0].Status), http.StatusBadRequest)
+		logger.Errorf("Status not OK: %v", resp.Results[0].Status)
+		return
 	}
+
+	logger.Infof("%#v", resp.Results[0].Value.Value())
+
+	w.WriteHeader(http.StatusOK)
+	// TODO: Should handle different type of return values and return JSON/other data
+	// types instead of plain text
+	rawRespBody := resp.Results[0].Value.Value()
+	rawRespBodyString := fmt.Sprintf("%v", rawRespBody)
+	respString := rawRespBodyString
+	instructionFuncName, shouldUsePythonCustomProcessing := deviceshifubase.CustomInstructionsPython[handlerInstruction]
+	logger.Infof("Instruction %v is custom: %v", handlerInstruction, shouldUsePythonCustomProcessing)
+	if shouldUsePythonCustomProcessing {
+		logger.Infof("Instruction %v has a python customized handler configured.\n", handlerInstruction)
+		respString = utils.ProcessInstruction(deviceshifubase.PythonHandlersModuleName, instructionFuncName, rawRespBodyString, deviceshifubase.PythonScriptDir)
+	}
+	fmt.Fprintf(w, "%v", respString)
+}
+
+type WriteRequest struct {
+	Value interface{} `json:"value"`
+}
```

Comment:
lowercase if not used anywhere else?

Reply:
lowercase cannot marshal json string into struct so